### PR TITLE
Notification to sysadmin when default JWT signing key is used

### PR DESCRIPTION
**Pull request by ViacheslavKlimov**
## Pull Request description

image

## General checklist

- [ ] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [ ] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [ ] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [ ] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

File: application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/DefaultJwtSettingsService.java [hunk 1]
```diff
@@ -124,6 +128,7 @@ public JwtSettings getJwtSettings(boolean forceReload) {
                         log.warn("WARNING: The platform is configured to use default JWT Signing Key. " +
                                 "This is a security issue that needs to be resolved. Please change the JWT Signing Key using the Web UI. " +
                                 "Navigate to \"System settings -> Security settings\" while logged in as a System Administrator.");
+                        notificationCenter.sendGeneralWebNotification(TenantId.SYS_TENANT_ID, new SystemAdministratorsFilter(), DefaultNotifications.jwtSigningKeyIssue.toTemplate());
```
Comment (smatvienko-tb) [inline application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/DefaultJwtSettingsService.java:131]: For me, it is better that JwtSettingsService know nothing about how to compose the notification.

It is much clear to have a simple call like 'notificationCenter.notifyJwtSettingsKeyIssue()'

File: application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/DefaultJwtSettingsService.java [hunk 2]
```diff
@@ -43,6 +46,7 @@ public class DefaultJwtSettingsService implements JwtSettingsService {
     private final AdminSettingsService adminSettingsService;
     @Lazy
     private final Optional<TbClusterService> tbClusterService;
+    private final NotificationCenter notificationCenter;
```
Comment (smatvienko-tb) [inline application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/DefaultJwtSettingsService.java:49]: The `JwtSettingsService` may run during install/upgrade phase. If notification center is not enabled under `install` profile, you should use the `Optional<notificationCenter>` similar to the 'TbClusterService'

File: application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/DefaultJwtSettingsService.java (hunk 1 above)
Comment (smatvienko-tb) [inline application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/DefaultJwtSettingsService.java:131]: As it is a high-security issue notification. Please, add to the test that 'notificationCenter' is being called.

Comment (ViacheslavKlimov) [reply application/src/main/java/org/thingsboard/server/service/security/auth/jwt/settings/DefaultJwtSettingsService.java:131]: JwtSettingsService only chooses needed notification from DefaultNotifications.
I don't want NotificationCenter to end up looking like MailService :)

File: application/src/main/java/org/thingsboard/server/service/notification/DefaultNotificationCenter.java [hunk 3]
```diff
@@ -165,16 +167,54 @@ public NotificationRequest processNotificationRequest(TenantId tenantId, Notific
...
+        NotificationTarget target = new NotificationTarget();
+        target.setTenantId(tenantId);
+        PlatformUsersNotificationTargetConfig targetConfig = new PlatformUsersNotificationTargetConfig();
+        targetConfig.setUsersFilter(recipients);
+        target.setConfiguration(targetConfig);
+
+        NotificationRequest notificationRequest = NotificationRequest.builder()
+                .tenantId(tenantId)
+                .template(template)
+                .targets(List.of(EntityId.NULL_UUID)) // TODO: refactor
```
[the remark this answers is not in the document]
Comment (ViacheslavKlimov) [reply application/src/main/java/org/thingsboard/server/service/notification/DefaultNotificationCenter.java:185]: Will update the comment for clarity

[the remark this answers is not in the document]
Comment (smatvienko-tb) [reply application/src/main/java/org/thingsboard/server/service/notification/DefaultNotificationCenter.java]: Even with clarified comments, finding the place to edit next time takes a lot of work.
It is better to point out some issue links or ticket numbers. Some identifiers that you can rely on in the future.

**Review by smatvienko-tb — APPROVED** [review]
Looks good to me! 
It will be nice to have a translation for non-ENG users

**Pull request merged**